packages/syft/src/syft/core/tensor/smpc/share_tensor.py

```diff
@@ -124,8 +124,8 @@ def __init__(
     def login_clients(parties_info: List[Party]) -> Any:
         clients = []
         for party_info in parties_info:
-            client = CACHE_CLIENTS.get(party_info, None)
             party_info.url = party_info.url.replace("localhost", "docker-host")
+            client = CACHE_CLIENTS.get(party_info, None)
```

Comment:
Here I reverse the order because in the other case we might have never been using the `cache` - I think this might bring an improvement

Reply:
Really great find, george,

Reply:
Apologies I was the one, who modified like that before

Reply:
Login is really costly george, This would give us a great improvement.